pom.xml

```diff
@@ -442,7 +442,7 @@
               -Dorg.apache.drill.exec.server.Drillbit.system_options="org.apache.drill.exec.compile.ClassTransformer.scalar_replacement=on"
               -Ddrill.test.query.printing.silent=true
               -Ddrill.catastrophic_to_standard_out=true
-              -XX:MaxPermSize=512M -XX:MaxDirectMemorySize=3072M
+              -XX:MaxPermSize=512M -XX:MaxDirectMemorySize=4096M
```

Comment:
Did all the regression tests pass ?  This change could have some impact on other tests (e.g., if was expecting an OOM, now gone).

Reply:
All the unit tests passed locally on my laptop. They also passed on our build server along with all of the functional and TPCH tests.

Reply:
It may be better to open a separate PR for DRILL-5926

Reply:
I could not properly test my PR because of a pre-existing sporadic test failure. This fix was necessary in order to test my changes. Errors with the builds should be fixed immediately, especially when they interfere with testing a change. So I will keep the fix in this PR.

Reply:
DRILL-5926 affects not only this PR but my PR  #1031 as well even though they are not related at all. At a minimum the change should be in a separate commit that refers to DRILL-5926 as it is a workaround and not an actual fix for the problem. In the future, the workaround may need to be reverted and if it is mixed with other changes, it will not be clear what needs to be reverted. Note that with other changes I would expect tests to require less memory, not more.

Reply:
I will move DRILL-5926 into a separate commit.

Reply:
This is a fix for a test that we don't really even need. We are attempting to verify that we can allocate a single value vector of 2 GB by doubling from a small amount. On the last allocation, we have a 1 GB vector doubling to 2 GB, so we temporarily need 3 GB.

But, vectors should never get this large. The batch size limitation project is doing work to limit vectors to the 16 MB Netty slab size.

For this reason, this is a workaround to a test for a feature that is actually a bug which we will be fixing in a later release.

Reply:
@paul-rogers I think there are a few choices available to us:

 1. Keep this change in a separate commit (already done) and revert it once the Batch Size Limitation work is done. The benefit is that the build will not be broken up until the Batch Size work is in. I can make a note on the Batch Size jira to revert this change as well.
 1. Remove this change and suffer with a broken build until the Batch Size work is complete. (not fun)
 1. Delete the offending tests instead as part of this PR.

I am in favor of option **1**. I am strongly against option **2** since ignoring build issues makes life much harder for everyone.

Please let me know what path forward you'd like to take.

Reply:
I'm in favor of 1 for now, then we can revert it, if needed, once the batch size work is available and we remove the offending test (option 3).

Reply:
I am in favor of 1 as well.

@paul-rogers One of the test that fails initially tries to allocate 2GB (without re-allocation). The test intermittently fails due to Pooled Allocator not releasing memory back to the system. I don't know if it is supposed to return memory back to the system when it is closed and whether it is supposed to be closed at all during the unit tests.

Reply:
Why not just retire the test. As explained above, no vector should ever allocate 2 GB unless we do some fixes to the Netty allocator. (Such a large vector will likely introduce issues in the network layer, but we don't have any user JIRAs to confirm this is, indeed, a problem.)